### PR TITLE
#187 Check `Capabilities` to render `PodDisruptionBudget` manifest

### LINKED ISSUE #187: `PodDisruptionBudget` resource should be optional - it now prevents deployment on K8s 1.20 and below
The template for the `PodDisruptionBudget` should only render if its resource type is available on the cluster.

### PR DESCRIPTION
**Pull request by vovtz**
Fixes #187

### CODE REVIEW

**Comment by joejulian** [comment]
The PDB is critical to prevent outages, split brain, and data loss. I think we should just prevent the chart from being used on k8s < 1.21 (which was merged before I saw this issue/PR). What do you think?

**Comment by vovtz** [comment]
For our current use case there is no HA requirement, we plan to use _Redpanda_ only to create ephemeral one-node ‘clusters’ that are used in component testing. In future we might decide to use _Redpanda_ in production too, but for now we use _Kafka_.

(BTW, we obviously shouldn’t run K8s versions that are EOL anyway - we urgently requested to upgrade.)

File: redpanda/templates/poddisruptionbudget.yaml [hunk 1]
```diff
@@ -35,3 +36,4 @@ spec:
       app.kubernetes.io/name: {{ template "redpanda.name" . }}
       app.kubernetes.io/instance: {{ .Release.Name | quote }}
   maxUnavailable: {{ .Values.statefulset.budget.maxUnavailable | int64 }}
+{{- end }}
```
Comment (joejulian) [inline redpanda/templates/poddisruptionbudget.yaml:39]: > For our current use case there is no HA requirement, we plan to use _Redpanda_ only to create ephemeral one-node ‘clusters’ that are used in component testing.

In that case, we should probably add a value under "below here be dragons" with a big warning about not using in production, then use that new value in an else, here.

something like:

```
yesyesyesno:
  pod_disruption_budget:
    enabled: true
```    

```suggestion
{{- else }}
  {{- if .Values.yesyesyesno.pod_disruption_budget.enabled }}
  fail "You must use Kubernetes 1.21+ for a production cluster. If you're not using this in production and you know why doing this is a bad idea, you can set yesyesyesno.pod_disruption_budget.enabled=false to continue."
  {{- end }
{{- end }}
```

(The parent key is not what I suggest, it's just a humorous placeholder)

**Review by joejulian — CHANGES_REQUESTED** [review]

File: redpanda/templates/poddisruptionbudget.yaml (hunk 1 above)
Comment (vovtz) [reply redpanda/templates/poddisruptionbudget.yaml:41]: I committed largely what you propose above, including an attempt for a more serious key name 😄

**Review by joejulian — APPROVED** [review]
Thank you. This seems like a good compromise.

**Comment by joejulian** [comment]
rebased

**Comment by joejulian** [comment]
Will rebase and merge after #202

**Pull request merged**